### ISSUE TITLE
Support Code Vision inlay hints for Rust code

### ISSUE THREAD

**Issue by Undin** [intellij-rust/intellij-rust#9427]
The platform has a feature called Code Vision which shows users some information about code: code authors, usages, related problems and inheritors

<img width="276" alt="image" src="https://user-images.githubusercontent.com/2539310/193036048-e17c6ab8-fa6b-4a6a-8c15-d92249e57a0a.png">

Currently, it works for Java but doesn't work for Rust

<img width="426" alt="Screenshot 2022-09-29 at 14 57 28" src="https://user-images.githubusercontent.com/2539310/193037641-f85994e3-03ad-49ce-b878-4ad25d32db41.png">

It would be nice to have this feature for Rust as well

* [ ] Code authors
* [ ] Usages
* [ ] Inheritors
* [ ] Related problems

**Comment by Kobzol** [comment]
Is there some API entry point or documentation available for this?

**Comment by sirgl** [comment]
@Kobzol Yeah, there is `com.intellij.codeInsight.hints.codeVision.CodeVisionProviderBase` for providers of inheritors and usages.
For code authors there is `com.intellij.codeInsight.hints.VcsCodeVisionLanguageContext` - it is very simple.
And related problems is very complicated thing and must be done for every language from scratch and I'm not sure if it will be easy for Rust. But for java its implementation here: `com.intellij.codeInsight.daemon.problems.pass.ProjectProblemCodeVisionProvider`

**Comment by Kobzol** [comment]
Are inheritors relevant for Rust? Maybe trait implementations?

**Comment by Undin** [comment]
@Kobzol yep. I think `inheritors` hints in Rust should show:
- number of trait implementations for trait itself
- number of implementations of associated items
- number of overrides for associated items with default implementations

Pretty the same as for Java interfaces
